docs/latest/tutorial/tutorial-scaffolding.md

```diff
@@ -0,0 +1,354 @@
+how all the pieces work together.
+:::
+
+## Adding some UI
+
+Before we can create a window for our application, we need to create the content that
+will be loaded into it. In Electron, each window displays web contents that can be loaded
+from either from a local HTML file or a remote URL.
+
+For this tutorial, you will be doing the former. Create an `index.html` file in the root
+folder of your project:
+
+```html title='index.html'
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8" />
+    <!-- https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP -->
+    <meta
+      http-equiv="Content-Security-Policy"
+      content="default-src 'self'; script-src 'self'"
+    />
+    <meta
+      http-equiv="X-Content-Security-Policy"
+      content="default-src 'self'; script-src 'self'"
+    />
+    <title>Hello from Electron renderer!</title>
+  </head>
+  <body>
+    <h1>Hello from Electron renderer!</h1>
+    <p>👋</p>
+  </body>
+</html>
+```
+
+Now that you have a web page, load it into an application window. To do so, paste
+the following code in your `main.js` file so we can disect it line by line:
+
+```js title='main.js'
+const { app, BrowserWindow } = require('electron')
+
+const createWindow = () => {
+  const win = new BrowserWindow({
+    width: 800,
+    height: 600,
+    webPreferences: {
+      sandbox: true,
+    }
+  })
+
+  win.loadFile('index.html')
+}
+
+app.whenReady().then(() => {
+  createWindow()
+})
+```
+
+In the first line we are `require`ing 2 Electron modules:
+
+- The [`app`][app] module, which controls your application's event lifecycle.
+- The [`BrowserWindow`][browser-window] module, which creates and manages application
+  windows.
+
+```js
+const { app, BrowserWindow } = require('electron')
+```
+
+:::warning
+ESM (i.e.: use `import` to load a module) is currently not supported in Electron directly.
+You can find more information about the state and chanllenges in
+[this issue](https://github.com/electron/electron/issues/21457).
+:::
+
+Then, `createWindow()` function loads `index.html` into a new `BrowserWindow`
+instance with a series of options:
+
+```js
+const createWindow = () => {
+  const win = new BrowserWindow({
+    width: 800,
+    height: 600,
+    webPreferences: {
+      sandbox: true
+    }
+  })
+
+  win.loadFile('index.html')
+}
+```
+
+:::tip
+You can find all the options available in the [`BrowserWindow` page][browser-window].
+:::
+
+In Electron, browser windows can only be created after the `app` module's
+[`ready`][app-ready] event is fired. You can wait for this event by using the
+[`app.whenReady()`][app-when-ready] API. Call `createWindow()` after `whenReady()`
+resolves its Promise. This is done in the last 3 lines of the code:
+
+```js
+app.whenReady().then(() => {
+  createWindow()
+})
+```
+
+At this point, your Electron application should successfully open a window that displays your
+web page!
```

Comment:
I realize now that we don't really introduce the concept of the renderer process at this point:

```suggestion
At this point, your Electron application should successfully open a window that displays your
web page! Each window created by your app will run in a separate process called a
**renderer**.
```